.github/workflows/workflow.yml

```diff
@@ -0,0 +1,65 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+# This workflow will install Deno and run tests across stable and canary builds on Windows, Ubuntu and macOS.
+# For more information see: https://github.com/denoland/setup-deno
+
+name: CodersCampWorkflow
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  build-and-run-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: install dependencies 
+        run: yarn install --immutable
+      - name: build api
+        run: yarn w api build
+      - name: build client
+        run: yarn w website build
+      - name: run affected tests 
+        run: yarn test --onlyChanged
+      
+  es-Lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: install dependencies 
+        run: yarn install --immutable
+      - name: Run ESLint
+        run: yarn eslint --cache
+  Typescript-compiler:
```

Comment:
It might be nice to define testing, linting, and type checking as separate workflows so they can run concurrently